files/Aboot/boot0.j2

```diff
@@ -169,37 +271,75 @@ extract_image() {
     ## Unpacking dockerfs delayed
     ## 1. when disk is vfat as it does not support symbolic link
     ## 2. when disk is small, expand it into ramfs during initrd
-    if [ "$rootfs_type" != "vfat" -a x"$docker_inram" != x"on" ]; then
-        mkdir -p "$image_path/{{ DOCKERFS_DIR }}"
+    if [ "$rootfs_type" != "vfat" ] && ! cmdline_has docker_inram=on; then
+        mkdir -p "$image_path/$docker_dir"
 
         if [ -n "$install" ]; then
             TAR_EXTRA_OPTION="--numeric-owner --warning=no-timestamp"
         fi
 
         ## extract docker archive
-        unzip -oqp "$swipath" "$dockerfs" | tar xzf - -C "$image_path/{{ DOCKERFS_DIR }}" $TAR_EXTRA_OPTION
+        unzip -oqp "$swipath" "$dockerfs" | tar xzf - -C "$image_path/$docker_dir" $TAR_EXTRA_OPTION
     else
         ## save dockerfs archive in the image directory
         unzip -oq "$swipath" "$dockerfs" -d "$image_path"
         info "Unpacking $dockerfs delayed to initrd because $target_path is $rootfs_type or docker_inram is on"
     fi
 
     ## remove installer since it's not needed anymore
-    info "Remove installer"
-    cleanup_swi_tmpfs
+    if $preserve_installer; then
+        info "Preserving installer under $installer_image_path"
+        mv "$swipath" "$installer_image_path"
+    else
+        info "Remove installer"
+        cleanup_swi_tmpfs "$(basename "$swipath")"
+    fi
 
     ## use new reduced-size boot swi
-    local swi_boot_path="flash:$image_name/{{ ABOOT_BOOT_IMAGE }}"
+    local swi_boot_path="flash:$image_name/$boot_image"
     update_boot_config SWI "$swi_boot_path"
     update_boot_config SWI_DEFAULT "$swi_boot_path"
 
     ## sync disk operations
     sync
 }
 
+extract_image_secureboot() {
+    info "Extracting necessary swi content"
+    mkdir "$image_path"
+    unzip -oq "$swipath" platform/firstime .imagehash -d "$image_path"
```

Comment:
first**t**ime?
Why we need this file? Why we don't need this file previously?

Reply:
I see the previous usage. Still this seems a typo as I mentioned above in **bold**

Reply:
Ah, I did saw that typo, nice catch!